### PR TITLE
planner: support sysdate behavior according to `sysdate_is_now`

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/31872

### What is changed and how it works?

1. introduce system variable `sysdate_is_now`
2. support change `sysdate` into `now` if the`sysdate_is_now` enabled.
3. support push `sysdate()` down to TiKV if `sysdate_is_now` is false

### Check List

Tests 

- [x] Unit test
- [x] Intergration test

### Release note

```release-note
Add system variable `sysdate_is_now`, default false. If it is true, `sysdate()` would be considered as an alias of `now()`
```

### CODE REVIEW

**Comment by Yisaer** [comment]
/hold cancel 

Due to tikv already support `sysdate` yet.

File: sessionctx/variable/sysvar.go [hunk 1]
```diff
@@ -1695,4 +1708,6 @@ const (
 	RandSeed1 = "rand_seed1"
 	// RandSeed2 is the name of 'rand_seed2' system variable.
 	RandSeed2 = "rand_seed2"
+	// TiDBSysdateIsNow is the name of the `sysdate_is_now` system variable
+	TiDBSysdateIsNow = "sysdate_is_now"
```
Comment (JmPotato) [inline sessionctx/variable/sysvar.go:1712]: This is not a TiDB-only system variable, I'd like to remove this prefix 🤔.

**Review by rebelice — COMMENTED** [review]
agree with JmPotato, rest LGTM

**Review by rebelice — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**